pkg/registry/extensions/deployment/strategy.go

```diff
@@ -77,6 +77,12 @@ func (deploymentStrategy) PrepareForUpdate(ctx api.Context, obj, old runtime.Obj
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
 
+	if newDeployment.Spec.Strategy.Type == extensions.RecreateDeploymentStrategyType {
```

Comment:
https://github.com/kubernetes/kubernetes/issues/39337 is for making strategies extensible.